geometry/rotation.hpp

```diff
@@ -20,16 +20,52 @@ template<typename FromFrame, typename ToFrame>
 std::ostream& operator<<(std::ostream& out,
                          Rotation<FromFrame, ToFrame> const& rotation);
 
+// |EulerAngles| and |CardanAngles| have values in binary-coded ternary
+// representing the sequence of rotation axes.
```

Comment:
If I declare constants, I have to use shifts rather than just chaining the numbers; we did that for permutations, with things like `(Y << X * 2)`, but there it made some sense (x maps to y); here it's not a mapping from XYZ to whichever sequence, so that would be confusing.
If you prefer that to the integer literals, I could declare a function that chains binary-coded ternary digits (e.g. something like `AxisConvention(X, Z, X)`.